zend_abstract_interface/hook/hook.h

```diff
@@ -0,0 +1,160 @@
+#ifndef ZAI_HOOK_H
+#define ZAI_HOOK_H
+// clang-format off
+#include <symbols/symbols.h>
+
+/* The Hook interface intends to abstract away the storage and resolution of hook targets */
+
+/* {{{ staging functions
+        Note: installation of hooks may occur after minit */
+bool zai_hook_minit(void);
+bool zai_hook_rinit(void);
+void zai_hook_activate(void);
+void zai_hook_clean(void);
+void zai_hook_rshutdown(void);
+void zai_hook_mshutdown(void); /* }}} */
+
+typedef bool (*zai_hook_begin)(zend_ulong invocation, zend_execute_data *frame, void *auxiliary, void *dynamic);
```

Comment:
What's invocation? The number of times it's been invoked or something? Everything I spot checked just cast it to void, so I'm wondering why it's here.